andela/octopus-bn-backend
# #171533958 Approve Request (By Requester’s Manager)

### What does this PR do?

- Add and authenticate ```PUT/api/v1/trips/:tripId/approve``` route
- Allow managers to approve trip requests that are only in **pending** mode
- Add swagger documentation for a feature
- Utilize available middlewares and validations with some slight changes
- Add French and English translation access


### How should this be manually tested?

- Run ```npm install ``` to setup necessary dependencies
- Run ``` npm run test ``` to see if everything is working as expected
- Run ``` npm run dev ``` to start a local server
- Head over to postman and send a PUT request on this route: ```localhost:3000/api/v1/trips/:tripId/approve```

### Implementation plan

[Approving trip request](https://docs.google.com/document/d/17JTeoICA4pstwlGaFs7jtIJNmXWvIohZqKCZU0NF3OE/edit?usp=sharing)

### Stories associated with PR

[#171533958](https://www.pivotaltracker.com/story/show/171533958)

### Screenshots if any
<img width="1119" alt="Screen Shot 2020-03-02 at 15 12 45" src="https://user-images.githubusercontent.com/51243858/75680172-a5d25c80-5c99-11ea-91f4-1f666a1bf1c3.png">
<img width="1096" alt="Screen Shot 2020-03-02 at 15 12 12" src="https://user-images.githubusercontent.com/51243858/75680184-ae2a9780-5c99-11ea-8f2b-ba02a1396ee5.png">
